src/main/java/frc/robot/commands/CrocodileDefaultCommand.java

```diff
@@ -3,33 +3,49 @@
 import java.util.function.BooleanSupplier;
 import java.util.function.DoubleSupplier;
 
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.XboxController;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.CrocodileSubsystem;
 
 public class CrocodileDefaultCommand extends CommandBase {
     private final CrocodileSubsystem subsystem;
-    private final DoubleSupplier motorSpeed;
-    private final BooleanSupplier slowMovement;
+    private final DoubleSupplier trigger;
+    private final XboxController driverController;
+    private boolean beamBreakTracker = false;
+    private final Debouncer debouncer;
 
-    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier motorSpeed,
-            BooleanSupplier slowMovement) {
+    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier trigger) {
         this.subsystem = subsystem;
-        this.motorSpeed = motorSpeed;
-        this.slowMovement = slowMovement;
+        this.trigger = trigger;
+        //TODO: get driver controller port
+        //controller to use for rumble
+        driverController = new XboxController(0);
+        debouncer = new Debouncer(0.5);
         addRequirements(subsystem);
     }
 
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        if (slowMovement.getAsBoolean()) {
-            subsystem.setMotorSpeed(.4);
-        } 
-        else if (Math.abs(motorSpeed.getAsDouble()) > 0.05){
-            subsystem.setMotorSpeed(motorSpeed.getAsDouble());
+        long startTime = 0;
+        subsystem.setIntakeSpeed(trigger.getAsDouble());
+        //sets rumble if beam break is broken for 0.5 seconds and is not already rumbling
+        if(debouncer.calculate(subsystem.getBeamBreak()) && !beamBreakTracker){
+            //get current time and set rumble
+            startTime = System.currentTimeMillis();
```

Comment:
Is there a reason that `startTime` is created up there and set down here, it can probably just be initialized to `System.currentTimeMillis`, like `long startTime = System.currentTimeMillis`.